src/main/webapp/app/overview/course-lectures/course-lecture-row.component.ts

```diff
@@ -1,24 +1,57 @@
-import { Component, HostBinding, Input, OnInit } from '@angular/core';
+import { Component, HostBinding, Input, OnChanges, SimpleChanges } from '@angular/core';
 import * as moment from 'moment';
 import { Moment } from 'moment';
 import { Course } from 'app/entities/course.model';
 import { ActivatedRoute, Router } from '@angular/router';
 import { Lecture } from 'app/entities/lecture.model';
+import { LectureUnitType } from 'app/entities/lecture-unit/lectureUnit.model';
 
 @Component({
     selector: 'jhi-course-lecture-row',
     templateUrl: './course-lecture-row.component.html',
     styleUrls: ['../course-exercises/course-exercise-row.scss'],
 })
-export class CourseLectureRowComponent implements OnInit {
+export class CourseLectureRowComponent implements OnChanges {
     @HostBinding('class') classes = 'exercise-row';
     @Input() lecture: Lecture;
     @Input() course: Course;
     @Input() extendedLink = false;
 
+    noOfExerciseUnits = 0;
+    noOfTextUnits = 0;
+    noOfVideoUnits = 0;
+    noOfAttachmentUnits = 0;
+
     constructor(private router: Router, private route: ActivatedRoute) {}
 
-    ngOnInit() {}
+    ngOnChanges(changes: SimpleChanges) {
+        for (const propName in changes) {
+            if (changes.hasOwnProperty(propName)) {
+                if (propName === 'lecture' && this.lecture && this.lecture.lectureUnits) {
```

Comment:
```suggestion
                if (changes.hasOwnProperty(propName) && propName === 'lecture' && this.lecture && this.lecture.lectureUnits) {
```

Reply:
Whats the idea behind first iterating through the changes and then checking if the changes contain such an attribute? Isn't this already given by the fact that we iterate through it?

Reply:
I have no idea. This is the recommended way according to :https://dev.to/nickraphael/ngonchanges-best-practice-always-use-simplechanges-always-1feg